libs/sdk-ui-dashboard/src/model/commands/layout.ts

```diff
@@ -0,0 +1,635 @@
+ * The new section will be placed at the desired index and will be empty by default.
+ *
+ * You may optionally specify the initial values of the section header and the items that will be in the new section.
+ *
+ * @param index - index to place the section at; -1 can be used as convenience to append a new section
+ * @param initialHeader - optionally specify specify header for the newly created section
+ * @param initialItems - optionally specify one or more items that the newly created section should include from the get-go
+ * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
+ *  events that will be emitted during the command processing
+ *
+ * @internal
+ */
+export function addLayoutSection(
+    index: number,
+    initialHeader?: IDashboardLayoutSectionHeader,
+    initialItems?: DashboardItemDefinition[],
+    correlationId?: string,
+): AddLayoutSection {
+    return {
+        type: "GDC.DASHBOARD.CMD.FL.ADD_SECTION",
+        correlationId,
+        payload: {
+            index,
+            initialHeader,
+            initialItems,
+        },
+    };
+}
+
+//
+//
+//
+
+/**
+ * @internal
+ */
+export interface MoveLayoutSection extends IDashboardCommand {
+    readonly type: "GDC.DASHBOARD.CMD.FL.MOVE_SECTION";
+    readonly payload: {
+        /**
+         * Index of the section to move.
+         *
+         * Index is zero-based.
+         */
+        readonly sectionIndex: number;
+
+        /**
+         * Index where the section should be moved.
+         *
+         * Index is zero-based. For convenience index of -1 means moving the item to the end of the section list.
+         */
+        readonly toIndex: number;
+    };
+}
+
+/**
+ * Creates the MoveLayoutSection command. Dispatching this command will result in move of the section located at `sectionIndex`
+ * to a new place indicated by `toIndex`.
+ *
+ * @param sectionIndex - index of section to move
+ * @param toIndex - the new index for the section
+ * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
+ *  events that will be emitted during the command processing
+ *
+ * @internal
+ */
+export function moveLayoutSection(
+    sectionIndex: number,
+    toIndex: number,
+    correlationId?: string,
+): MoveLayoutSection {
+    return {
+        type: "GDC.DASHBOARD.CMD.FL.MOVE_SECTION",
+        correlationId,
+        payload: {
+            sectionIndex,
+            toIndex,
+        },
+    };
+}
+
+//
+//
+//
+
+/**
+ * @internal
+ */
+export interface RemoveLayoutSection extends IDashboardCommand {
+    readonly type: "GDC.DASHBOARD.CMD.FL.REMOVE_SECTION";
+    readonly payload: {
+        /**
+         * Index of section to remove.
+         *
+         * Zero based. For convenience -1 can be used to remove the last section.
```

Comment:
Is there a usecase for this? Or is it just ot keep this consistent with move/reset/add etc?

Reply:
I mean the -1 part

Reply:
yep; the idea was to simplify 'remove last row'.